### PR TITLE
Start peeling off methods from the top of the NodeConstructor construct method

### PR DESCRIPTION
**Pull request by thecoop**
Create some methods for the first operations at the top of NodeConstructor

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/node/NodeConstruction.java [hunk 1]
```diff
@@ -399,80 +404,104 @@ private void construct(Environment initialEnvironment, NodeServiceProvider servi
...
+    }
 
-        final ThreadPool threadPool = new ThreadPool(settings, executorBuilders.toArray(new ExecutorBuilder<?>[0]));
+    private ThreadPool createThreadPool(Settings settings) throws IOException {
+        ThreadPool threadPool = new ThreadPool(
+            settings,
+            pluginsService.flatMap(p -> p.getExecutorBuilders(settings)).toArray(ExecutorBuilder<?>[]::new)
+        );
         resourcesToClose.add(() -> ThreadPool.terminate(threadPool, 10, TimeUnit.SECONDS));
-        final ResourceWatcherService resourceWatcherService = new ResourceWatcherService(settings, threadPool);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/node/NodeConstruction.java:416]: This constructor schedules some tasks - do these explicitly need to be without the HeaderContext set below?

File: server/src/main/java/org/elasticsearch/node/NodeConstruction.java [hunk 2]
```diff
@@ -241,8 +241,18 @@ static NodeConstruction prepareConstruction(
...
         List<Closeable> closeables = new ArrayList<>();
         try {
+            Settings envSettings = Settings.builder()
+                .put(initialEnvironment.settings())
+                .put(Client.CLIENT_TYPE_SETTING_S.getKey(), "node")
+                .build();
+
             NodeConstruction constructor = new NodeConstruction(closeables);
-            constructor.construct(initialEnvironment, serviceProvider, forbidPrivateIndexSettings);
+            Settings settings = constructor.createEnvironment(initialEnvironment, envSettings, serviceProvider);
```
Comment (ldematte) [inline server/src/main/java/org/elasticsearch/node/NodeConstruction.java:250]: It's a good step and I see you are using returned types to enforce relationship and order between calls; however I find a bit confusing to have this function have both side effects (sets environment and pluginService) and return something. 
It's OK for now as it minimizes change and makes easier to see the PR does not change behaviour, but I would consider a better alternative for the final refactoring step.

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/node/NodeConstruction.java:250]: There's going to be lots of steps to this refactoring - this is only the first

File: server/src/main/java/org/elasticsearch/node/NodeConstruction.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/node/NodeConstruction.java:434]: It's here from https://github.com/elastic/elasticsearch/pull/54993, looks like it was just moved as early as possible

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by ldematte — APPROVED** [review]
LGTM 
I see that settings validation and creation of additionalSettings/SettingsModule are moved before the creation of TaskManager and ResourceWatcherService, but it seems they are not interfering in any way.

**Comment by ldematte** [comment]
Only one doubt: it seems that CI is not running? I would make sure CI test suites run before merging

**Pull request merged**